enterprise/server/test/integration/remote_execution/remote_execution_test.go

```diff
@@ -352,13 +359,37 @@ func TestSimpleCommand_RunnerReuse_PoolSelectionViaHeader_RoutesCommandToSameExe
 
 	require.Equal(t, 0, res.ExitCode)
 
-	cmd = rbe.Execute(&repb.Command{
-		Arguments: []string{"stat", "foo.txt"},
-		Platform:  platform,
-	}, opts)
-	res = cmd.Wait()
+	// Run another command that tries to read "foo.txt" written by a previous
```

Comment:
Instead of blindly retying, what do you think about adding a prometheus metric for successful recycles and waiting on that?

Reply:
SG -- done